assets/src/modules/config/BaseLayer.js

```diff
@@ -926,6 +935,7 @@ export class BaseLayersConfig {
 
```

Comment:
I propose to do so instead of your proposed code.
```suggestion
        const default_background_color_index = options.hasOwnProperty('default_background_color_index') ? options.default_background_color_index : -1;
        if (isNaN(default_background_color_index)) {
            throw new ValidationError('The default_background_color_index parameter is not a number!');
        }
        if (default_background_color_index !== -1) {
            if (default_background_color_index < names.length) {
                names.splice(default_background_color_index, 0, 'project-background-color');
            } else {
                names.push('project-background-color');
            }
        }
```